### PR TITLE
isomer treatment in pyne.endf

### LINKED ISSUE #682: isomer treatment in pyne.endf
I ran into trouble reading an ENDF formatted file containing data for an Isomer (e.g. Pm-148M, https://www-nds.iaea.org/public/download-endf/ENDF-B-VII.1/n/n_6153_61-Pm-148M.zip) using the class provided by pyne.endf. The data of the endf file is stored using index 611480000, however it should be 611480001. I tried to write a fix, but that broke other tests. So here is some explanation:

I think the excited state should be triggered by the value of the LISO entry in the first part of the endf file. The manual says "LISO: Isomeric state number. The ground state is indicated by LISO=0.  [...]" 

The data structure of the beginning of 451 from ENDF Manual looks like this:
```
[MAT, 1,451/ ZA, AWR, LRP, LFI, NLIB, NMOD]HEAD
[MAT, 1,451/ ELIS, STA, LIS, LISO, 0, NFOR]CONT
[MAT, 1,451/ AWI, EMAX, LREL, 0, NSUB, NVER]CONT
[MAT, 1,451/ TEMP, 0.0, LDRV, 0, NWD, NXC]CONT
```

The file I used is an ENDF File downloaded from IAEA database (). It has LISO=1, but the current code fails to use the correct nuclide id. I add a PR with some example ENDF snippet + test in a few minutes. Tracing back the issue in https://github.com/pyne/pyne/blob/develop/pyne/endf.pyx it seems to be a problem in the following code:

```
line = fh.readline()
mat_id = int(line[66:70].strip() or -1)
# originally in a float version of ZZAAA.M, ie 94242.1
nuc = cpp_nucname.id(<int> (endftod(line[:11])*10))
# Make a new dict in self.structure to contain the material data.
```

Apparently the code assumes that the isomeric state is store as a decimal in the ZA entry, which is not the case for all the files at ENDF-B-VII.1 I looked at. So I made a fix suitable for this case (second PR).

However, there is a test snippet currently in the test file that has a line, that actually uses the implemented way to distinguish isomers.
```1.003100+3 2.098312+0 1 0 0 1 131 1451 1```

With my fix, this fails, because the test snippet does not have a LISO=1 entry. I'm not sure if this is an inconsistency in the ENDF definition, or just a wrong posed test. I would happy to fix it, either
a) add a check to pyne.endf to make both ways work
b) correct snippet in test_endf
but I am not sure what would be the correct solution.

### PR DESCRIPTION
**Pull request by ohnemax**
This could fix #682 , but it breaks a test. See the issue for further description.

Probably the second commit is not needed, it just resets file pointer to the second line of the MT=451 Head after the new operation. This is the same position as it was before.

### CODE REVIEW

**Comment by crbates** [comment]
This looks good to me @ohnemax . @paulromano do you know of any endf datasets where the zzzaaa.m form is used but LISO isn't used we should be concerned about?

**Comment by paulromano** [comment]
@crbates I'm not aware of any, but if there were such a dataset I would consider it to be a bug.

File: pyne/endf.pyx [hunk 1]
```diff
@@ -567,7 +577,14 @@ class Library(rx.RxLib):
...
         isotope_flags = self._get_cont(['ZAI','ABN',0,'LFW','NER',0],
                                        isotope_data[0])
-        nuc_i = nucname.id(int(isotope_flags['ZAI']*10))
+        # according to endf manual, there is no specification for metastable states in ZAI
+        # if we have a LIS0 != 0 we add the state to all isotopes
+        if(self.structure[mat_id]['matflags']['LIS0'] == 0):
+            nuc_i = nucname.id(int(isotope_flags['ZAI']*10))
+        else:
+            nuc_i = nucname.id(int(isotope_flags['ZAI']*10+self.structure[mat_id]['matflags']['LIS0']))
+#        nuc_i = nucname.id(int(isotope_flags['ZAI']*10))
```
Comment (crbates) [inline pyne/endf.pyx]: this commented line should be deleted as it's in the git history

File: pyne/endf.pyx [hunk 2]
```diff
@@ -567,7 +577,14 @@ class Library(rx.RxLib):
         """
         isotope_flags = self._get_cont(['ZAI','ABN',0,'LFW','NER',0],
                                        isotope_data[0])
-        nuc_i = nucname.id(int(isotope_flags['ZAI']*10))
+        # according to endf manual, there is no specification for metastable states in ZAI
+        # if we have a LIS0 != 0 we add the state to all isotopes
+        if(self.structure[mat_id]['matflags']['LIS0'] == 0):
+            nuc_i = nucname.id(int(isotope_flags['ZAI']*10))
+        else:
+            nuc_i = nucname.id(int(isotope_flags['ZAI']*10+self.structure[mat_id]['matflags']['LIS0']))
```
Comment (crbates) [inline pyne/endf.pyx]: Style issue here. PEP8 suggests spaces around `+` as it is the lowest priority operator see https://www.python.org/dev/peps/pep-0008/#other-recommendations

**Comment by crbates** [comment]
@ohnemax there are a few minor style issues I missed yesterday.

**Comment by scopatz** [comment]
ping @ohnemax

**Pull request merged**